lib/routes/account.js

```diff
@@ -139,10 +139,32 @@ module.exports = (
           return db.deleteAccount(emailRecord)
         }
 
-        function ignoreUnknownAccountError (err) {
-          if (err.errno !== error.ERRNO.ACCOUNT_UNKNOWN) {
-            throw err
+        function checkAccountError (err) {
+          if (err.errno === error.ERRNO.ACCOUNT_UNKNOWN) {
+            // Even though the email is not on the account table, it could be on the emails table.
+            // Delete the unverified secondary email if it exists and proceed with account creation.
+            return deleteSecondaryEmailIfUnverified()
           }
+
+          throw err
+        }
+
+        function deleteSecondaryEmailIfUnverified() {
+          return db.getSecondaryEmail(email)
+            .then((secondaryEmailRecord) => {
+              // Currently, users can not create an account from a verified
+              // secondary email address
+              if (secondaryEmailRecord.isVerified) {
+                throw error.verifiedSecondaryEmailAlreadyExists()
```

Comment:
The reason behind the question above -  Can we avoid revealing that some email is secondary vs primary? (unless we really needed it for some front-end features)